include/net/net_ip.h

```diff
@@ -128,14 +131,65 @@ struct sockaddr_in_ptr {
 	struct in_addr		*sin_addr;     /* IPv4 address */
 };
 
-#if defined(CONFIG_NET_IPV4) && !defined(CONFIG_NET_IPV6)
+struct sockaddr_ll {
+	sa_family_t sll_family;   /* Always AF_PACKET        */
+	u16_t       sll_protocol; /* Physical-layer protocol */
+	int         sll_ifindex;  /* Interface number        */
+	u16_t       sll_hatype;   /* ARP hardware type       */
+	u8_t        sll_pkttype;  /* Packet type             */
+	u8_t        sll_halen;    /* Length of address       */
+	u8_t        sll_addr[8];  /* Physical-layer address  */
+};
+
+struct sockaddr_ll_ptr {
+	sa_family_t sll_family;   /* Always AF_PACKET        */
+	u16_t       sll_protocol; /* Physical-layer protocol */
+	int         sll_ifindex;  /* Interface number        */
+	u16_t       sll_hatype;   /* ARP hardware type       */
+	u8_t        sll_pkttype;  /* Packet type             */
+	u8_t        sll_halen;    /* Length of address       */
+	u8_t        *sll_addr;    /* Physical-layer address  */
+};
+
+/* Packet types.  */
+#define PACKET_HOST         0     /* To us            */
+#define PACKET_BROADCAST    1     /* To all           */
+#define PACKET_MULTICAST    2     /* To group         */
+#define PACKET_OTHERHOST    3     /* To someone else  */
+#define PACKET_OUTGOING     4     /* Originated by us */
+#define PACKET_LOOPBACK     5
+#define PACKET_FASTROUTE    6
+
+#if defined(CONFIG_NET_IPV4)
+#undef NET_SOCKADDR_MAX_SIZE
```

Comment:
I'd say it's a bit strange to have these #undef's. In such way, we can redefine these values in loop, and won't have a hint where the issues happens. Can't we just define these values consistently in one chunk of preprocessor code, without #ifdef's. @jukkar, @tbursztyka: Please have a look, I'd be concerned in your place.

Reply:
The undef's are there in order to avoid preprocessor error if multiple protocols are defined. The idea is that the largest size sockaddr is the last one.

Reply:
> The idea is that the largest size sockaddr is the last one.

I'm not sure how obvious is that idea. It's definitely not documented in a code comment before that chunk, so someone adding a new sockaddr struct can easily stuff it in a random place in sequence.

Reply:
Adding comments are enough? Is there anything to do further?

Reply:
I'd say yes, if it's documented in a comment before that block, it's fine (I assume @jukkar introduced it in his initial version).

Reply:
> I assume @jukkar introduced it in his initial version

yes.

Reply:
> I'd say yes, if it's documented in a comment before that block, it's fine

Done.